### ISSUE TITLE
Switching to atl, it starts checking .js files and finding errors.

### ISSUE THREAD

**Issue by miracle2k** [s-panferov/awesome-typescript-loader#478]
My loaders are configured like this:

```
    {
        test: /\.tsx?$/,
        use: [
          'babel-loader',
          'awesome-typescript-loader'
        ],
        exclude: /node_modules/,
      },
      {
        test: /\.jsx?$/,
        loader: 'babel-loader',
        include: path.join(__dirname, 'src')
      },
```

This works fine if I use the regular ``ts-loader``. If I switch to atl, I am getting a whole lot of error messages, such as:

```
[at-loader] ./src/utils/i18n/index.js:111:61
    TS1136: Property assignment expected.

[at-loader] ./src/utils/shapes.js:11:13
    TS8008: 'type aliases' can only be used in a .ts file.

[at-loader] ./src/HoldAndPrefetch.js:53:3
    TS7027: Unreachable code detected.
```

Some of them seem to be due to features such as the ``import()`` call, that is only supported in later TS versions. Some of them seem to be because of flow.js annotations, which babel usually strips. Note also "Unreachable code".

Basically, it seems to me that with ts-loader, my .js files never go through typescript, but with atl, typescript tries to parse them. Why might that be?
